### DISCUSSION TITLE
[Snippet] Using next/font with Tailwind

### DISCUSSION THREAD

**Discussion by agusterodin** [vercel/next.js#42023] [Show and tell]
tailwind.config.js
```
module.exports = {
  content: ['./pages/**/*.{js,ts,jsx,tsx}', './components/**/*.{js,ts,jsx,tsx}'],
  theme: {
    fontFamily: {
      display: 'var(--exo-font)',
      sans: 'var(--open-sans-font)'
    }
}
```

Fonts.tsx

```
import { Exo, Open_Sans as OpenSans } from '@next/font/google'

const exo = Exo({ display: 'swap', fallback: ['Helvetica Neue', 'Helvetica', 'sans-serif'], subsets: ['latin'] })
const openSans = OpenSans({ display: 'swap', fallback: ['Helvetica Neue', 'Helvetica', 'sans-serif'], subsets: ['latin'] })

export default function Fonts() {
  return (
    <style jsx global>{`
      :root {
        --exo-font: ${exo.style.fontFamily};
        --open-sans-font: ${openSans.style.fontFamily};
      }
    `}</style>
  )
}
```

app.tsx
```
import { AppProps } from 'next/app'
import Head from 'next/head'
import Fonts from 'components/Fonts'
import 'styles.css'

export default function App({ Component, pageProps }: AppProps) {
  return (
    <>
      <Head>
        <title>Website Name</title>
      </Head>
      <Fonts />
      <Component {...pageProps} />
    </>
  )
}
```

**Comment by SalGnt** [comment]
Nice! I think you can get rid of the `Fonts.tsx` component and use of the built in [`variable`](https://nextjs.org/docs/api-reference/next/font#variable-1).

**Reply by kyuumeitai** [reply]
Nice catch. Although the `Fonts.tsx` component allows to specify fallbacks, in that matter, what method is the best?

**Reply by SalGnt** [reply]
You can still use `fallback` provided by `@next/font/*`. The only thing you'll skip is the manual creation of the variables performed in `Fonts.tsx`.

**Reply by itzomen** [reply]
@kyuumeitai can you please provide an example. I was not able to set up my fonts globally using the `variable`

The `Fonts.tsx` works well.

**Reply by SalGnt** [reply]
> @kyuumeitai can you please provide an example. I was not able to set up my fonts globally using the `variable`
> 
> The `Fonts.tsx` works well.

You can take a look at [this](https://alvar.dev/blog/tailwindcss-with-next-font) article wrote by @alvarlagerlof.

**Reply by SalGnt** [reply]
You can also find an an example [here](https://beta.nextjs.org/docs/optimizing/fonts#with-tailwind-css).

**Reply by itzomen** [reply]
Ok thanks, btw I couldn't get that to work in app.tsx
And I forgot to mention I'm using good old css.
Well I don't see any caveat with Fonts.tsx so I will stick to that for now

**Reply by ghost** [reply]
> @kyuumeitai can you please provide an example. I was not able to set up my fonts globally using the `variable`
> 
> The `Fonts.tsx` works well.

`Font.tsx` works but i get the _client/server mismatch [error](https://github.com/vercel/next.js/discussions/42023#discussioncomment-4027538)_ when i try to upload multiple local fonts to a font family as shown in the docs [here](https://beta.nextjs.org/docs/optimizing/fonts#local-fonts). And while the referenced solution works, it's a bit too ugly and causes subsequent unwanted behavior. 

[Variable](https://nextjs.org/docs/api-reference/next/font#variable-1) didn't work initially but it turned out i forgot to add ``className={`${Font1.variable} ${Font2.variable}`}`` in `layout.tsx` to my root tag.

Worked like a charm after i added the `className` and it seems like the way to go for now.

**Comment by kyuumeitai** [comment]
If by any chance you have the client/server mismatch error: Warning: Text content did not match. Server ... Client ... , you can apply this solution on the `Fonts.tsx` file: 

https://github.com/vercel/next.js/issues/3108#issuecomment-446559167
